src/EA.Iws.RequestHandlers/NotificationMovements/BulkReceiptRecovery/ReceiptRecoveryRecoveryOnlyRule.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace EA.Iws.RequestHandlers.NotificationMovements.BulkReceiptRecovery
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Core.Movement;
+    using Core.Movement.BulkReceiptRecovery;
+    using Core.Rules;
+    using Domain.Movement;
+
+    public class ReceiptRecoveryRecoveryOnlyRule : IReceiptRecoveryContentRule
+    {
+        private readonly IMovementRepository repo;
+
+        public ReceiptRecoveryRecoveryOnlyRule(IMovementRepository repo)
+        {
+            this.repo = repo;
+        }
+
+        public async Task<ReceiptRecoveryContentRuleResult<ReceiptRecoveryContentRules>> GetResult(List<ReceiptRecoveryMovement> movements, Guid notificationId)
+        {
+            var actualMovements = await repo.GetAllMovements(notificationId);
+
+            List<int> shipments = new List<int>();
+            MessageLevel result = MessageLevel.Success;
+
+            foreach (var movement in movements.Where(p => p.MissingReceivedDate))
```

Comment:
As noted on the other rule, please use `ReceivedDate.HasValue` rather than the `MissingReceivedDate` property.